core/app/models/spree/tax/tax_helpers.rb

```diff
@@ -17,9 +17,9 @@ module TaxHelpers
       #
       # For further discussion, see https://github.com/spree/spree/issues/4397 and https://github.com/spree/spree/issues/4327.
       def applicable_rates(order)
-        order_zone_tax_categories = rates_for_order_zone(order).map(&:tax_category)
+        order_zone_tax_categories = rates_for_order_zone(order).flat_map(&:tax_categories)
```

Comment:
We should probably add `.includes(:tax_categories)` to the scopes in `rates_for_order_zone` and `rates_for_default_zone` yeah?